src/sagemaker/async_inference/async_inference_response.py

```diff
@@ -87,8 +87,31 @@ def get_result(
         return self._result
 
     def _get_result_from_s3(self, output_path, failure_path):
+        """Retrieve output based on the presense of failure_path"""
```

Comment:
Lgtm - synced with Praneeth offline as well; we'll track a backlog item to go back to SageMaker Python SDK again and add Integ tests for edge case where customers created Async endpoint without failure path and loaded in Python SDK for invocation usage.